### PR TITLE
Ensure the access token is initialized when constructing a client

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
CHANGELOG_BEGIN
- [Java Client] Ensure the access token is initialized when using a
deprecated constructor.
CHANGELOG_END

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags, if appropriate
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: language-support/java/bindings-rxjava/src/main/java/com/daml/ledger/rxjava/DamlLedgerClient.java [hunk 1]
```diff
@@ -18,12 +18,16 @@
  * A {@link LedgerClient} implementation that connects to
  * an existing Ledger and provides clients to query it. To use the {@link DamlLedgerClient}:
  * <ol>
- *     <li>Create an instance of {@link DamlLedgerClient} using {@link DamlLedgerClient#forLedgerIdAndHost(String, String, int, Optional)},
- *     {@link DamlLedgerClient#forHostWithLedgerIdDiscovery(String, int, Optional)} or {@link DamlLedgerClient#DamlLedgerClient(Optional, ManagedChannel)}</li>
- *     <li>When ready to use the clients, call the method {@link DamlLedgerClient#connect()} to initialize the clients
- *     for that particular Ledger</li>
+ *     <li>Create an instance of a {@link Builder} using {@link DamlLedgerClient#newBuilder(String, int)}
+ *     <li>Specify and expected ledger identifier, {@link SslContext}, and/or access token, depending on your needs</li>
```
Comment (gerolf-da) [inline language-support/java/bindings-rxjava/src/main/java/com/daml/ledger/rxjava/DamlLedgerClient.java]:
```suggestion
 *     <li>Specify the expected ledger identifier, {@link SslContext}, and/or access token, depending on your needs</li>
```

File: language-support/java/bindings-rxjava/src/main/java/com/daml/ledger/rxjava/DamlLedgerClient.java [hunk 2]
```diff
@@ -147,6 +151,7 @@ private DamlLedgerClient(@NonNull NettyChannelBuilder channelBuilder, @NonNull O
     public DamlLedgerClient(Optional<String> expectedLedgerId, @NonNull ManagedChannel channel) {
         this.channel = channel;
         this.expectedLedgerId = expectedLedgerId.orElse(null);
+        this.accessToken = Optional.empty();
```
Comment (gerolf-da) [inline language-support/java/bindings-rxjava/src/main/java/com/daml/ledger/rxjava/DamlLedgerClient.java]: Do you think it's worth capturing this in a test?

File: language-support/java/bindings-rxjava/src/main/java/com/daml/ledger/rxjava/DamlLedgerClient.java (hunk 1 above)
Comment (gerolf-da) [reply language-support/java/bindings-rxjava/src/main/java/com/daml/ledger/rxjava/DamlLedgerClient.java]: Or maybe `an`?

**Comment by stefanobaghino-da** [comment]
Sorry, my bad, I should go lighter on the `automerge`. I'll address your comments in separate PR.

File: language-support/java/bindings-rxjava/src/main/java/com/daml/ledger/rxjava/DamlLedgerClient.java (hunk 1 above)
Comment (stefanobaghino-da) [reply language-support/java/bindings-rxjava/src/main/java/com/daml/ledger/rxjava/DamlLedgerClient.java]: Addressed by https://github.com/digital-asset/daml/pull/3793

File: language-support/java/bindings-rxjava/src/main/java/com/daml/ledger/rxjava/DamlLedgerClient.java (hunk 2 above)
Comment (stefanobaghino-da) [reply language-support/java/bindings-rxjava/src/main/java/com/daml/ledger/rxjava/DamlLedgerClient.java]: (Not) addressed by https://github.com/digital-asset/daml/pull/3793
This constructor is deprecated and our Scala tests are compiled with fatal warning (which unfortunately cannot be adjusted to whitelist given warnings).

**Pull request merged**